alibi_detect/od/llr.py

```diff
@@ -144,6 +144,10 @@ def fit(self,
             Callbacks used during training.
         """
         input_shape = X.shape[1:]
+        optimizer = optimizer() if isinstance(optimizer, type) else optimizer
+        # Separate into two separate optimizers, one for semantic model and one for background model
+        optimizer_s = optimizer
+        optimizer_b = optimizer.__class__.from_config(optimizer.get_config())
```

Comment:
This detector is a special case. The optimizer is called on two separate models during fit, which now leads to errors. To fix the optimizer is cloned so that separate optimizers are used for each model.

Reply:
I assume we don't have to worry about potentially sharing optimizer state between the two models or diverging optimizer states @arnaudvl ?

Reply:
No they don't have to share state.